tools/serve/serve.py

```diff
@@ -827,12 +829,15 @@ def run(**kwargs):
             servers = start(config, build_routes(config["aliases"]), **kwargs)
 
             try:
-                while any(item.is_alive() for item in iter_procs(servers)):
+                while all(item.is_alive() for item in iter_procs(servers)):
                     for item in iter_procs(servers):
                         item.join(1)
             except KeyboardInterrupt:
                 logger.info("Shutting down")
 
+        for item in iter_procs(servers):
+            logger.info("Status of %s:\t%s" % (item.name, "running" if item.is_alive() else "not running"))
```

Comment:
If `item.name` is `None` here the message won't be great. Is that impossible? If not, maybe just require `name` after all?

Reply:
Thanks for pointing this out. Including the port number in the name is a litle redundant since that's already defined via an instance property. Because that property *is* available on all instances, we can name the process with something slightly useful even when additional information (i.e. the scheme) is not specified.